src/dscanner/analysis/run.d

```diff
@@ -1350,6 +1346,12 @@ MessageSet analyzeDmd(string fileName, ASTCodegen.Module m, const char[] moduleN
 			config.redundant_storage_classes == Check.skipTests && !ut
 		);
 
+	if (moduleName.shouldRunDmd!(NumberStyleCheck!ASTCodegen)(config))
```

Comment:
Why is ASTCodegen passed here since we are only using the lexer?

Reply:
Because the BaseAnalyzerDmd follows an inheritance chain starting with the parent class ParseTimeVisitor!ASTCodegen . Should ASTBase struct be used instead?

Reply:
https://github.com/Dlang-UPB/D-scanner/issues/96